src/components/challenges/challenge.tsx

```diff
@@ -38,8 +39,11 @@ export default function Challenge(props: ChallengeProps) {
   const completed = state[id]?.possibleOptions &&
     [...state[id].possibleOptions].every(entry => state[id]?.selectedOptions.has(entry));
 
-  // checkbox and multiple-choice:
+  // Checkbox and Multiple Choice:
   const isAnswerCheckable = ['checkbox', 'multiple-choice'].includes(challengeType);
+
+  // Paragraph:
+  const isMultilineEntry = challengeType === 'paragraph';
```

Comment:
`Challenge` is currently a mess and warrants being split up into multiple components based on challenge type. I'm letting it be a mess for now, to see if a good low-repetition design emerges.